docs/OpenSSLSupport.md

```diff
@@ -0,0 +1,196 @@
+# Open Enclave Support for OpenSSL
+
+The OpenSSL on OE is configured with following options
```

Comment:
Follow-up: We need to get all of these options and header support reviewed prior to taking this out of experimental:
- some of them don't seem consistent, for example, no-dso should be used with no-engine, so should probably be added until we figure out the engine support.
- some require explanations with context, such as `--with-rand-seed=none`. It's an okay choice since the default rand implementation just doesn't work, but nothing makes the connection here.
- there are probably many flags we want to add to minimize TCB and support surface. For example, no-psk is probably a good one, and a lot of this could be driven by comparing the parity functional set with the mbedTLS config we currently ship and has been previously reviewed.

Reply:
Thanks for the comments.  I removed the `--with-rand-seed=none` in this PR given that it's not necessary as you mentioned.
I'll investigate the rest options as the follow-up.

Reply:
Updated: It turns out the `--with-rand-seed=none` can't be removed because it introduces other dependencies that we don't support. I'll leave the investigation as part of the follow-up.

Reply:
That's fine, all of the follow-up items are tagged as such because I expect that the subsequent PRs in this train of work will address them; thanks for documenting your finding here though, it's always helpful!